off_chain_data/application-go/parser/block.go

```diff
@@ -0,0 +1,101 @@
+package parser
+
+import (
+	"fmt"
+
+	"github.com/hyperledger/fabric-protos-go-apiv2/common"
+	"google.golang.org/protobuf/proto"
+)
+
+type Block struct {
+	block              *common.Block
+	cachedTransactions []*Transaction
+}
+
+func ParseBlock(block *common.Block) *Block {
+	return &Block{block, nil}
+}
+
+func (b *Block) Number() uint64 {
+	return b.block.GetHeader().GetNumber()
+}
+
+func (b *Block) Transactions() ([]*Transaction, error) {
+	if b.cachedTransactions != nil {
+		return b.cachedTransactions, nil
+	}
+
+	funcName := "Transactions"
+	envelopes, err := b.unmarshalEnvelopesFromBlockData()
+	if err != nil {
+		return nil, fmt.Errorf("in %s: %w", funcName, err)
+	}
+
+	commonPayloads, err := b.unmarshalPayloadsFrom(envelopes)
+	if err != nil {
+		return nil, fmt.Errorf("in %s: %w", funcName, err)
+	}
+
+	payloads, err := b.parse(commonPayloads)
+	if err != nil {
+		return nil, fmt.Errorf("in %s: %w", funcName, err)
+	}
+
+	b.cachedTransactions = b.createTransactionsFrom(payloads)
+
+	return b.cachedTransactions, nil
+}
+
+func (b *Block) unmarshalEnvelopesFromBlockData() ([]*common.Envelope, error) {
```

Comment:
When it's a method on Block, it seems like repetition to say "FromBlockData". Perhaps keep it simple?

```suggestion
func (b *Block) unmarshalEnvelopes() ([]*common.Envelope, error) {
```

Reply:
Agree. Done.